tests/utils/test_core.py

```diff
@@ -4,11 +4,11 @@
 
 import numpy as np
 import pandas as pd
+from pandas.api.types import infer_dtype
```

Comment:
Let's make the tests also run without pandas installed so that we can run the whole test suite once with pandas installed and once without. Prevents us from accidentally reintroducing a hard dependency again in the future